### PR TITLE
[refactor] delete hourly/daily/weekly/monthly schedule decorators, PartitionScheduleDefinition, build_schedule_from_partition

### PR DESCRIPTION
**Pull request by smackesey**
## Summary & Motivation

Internal companion PR: https://github.com/dagster-io/internal/pull/5208

Delete legacy schedule decorators,`PartitionScheduleDefinition`, `build_schedule_from_partitions`.

Mostly this involved converting uses of these decorators into regular `@schedule` calls. Needed to adjust a lot of tests for compatibility with the newer APIs. Some tests targeting `PartitionScheduleDefinition` were removed entirely.

## How I Tested These Changes

Existing test suite.

### CODE REVIEW

**Comment by smackesey** [comment]
Current dependencies on/for this PR:
* master
  * **PR #11996** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11996" target="_blank">Graphite</a> 
    * **PR #11953** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11953" target="_blank">Graphite</a> 
      * **PR #11952** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11952" target="_blank">Graphite</a> 
        * **PR #12484** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12484" target="_blank">Graphite</a> 
          * **PR #12189** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12189" target="_blank">Graphite</a> 
            * **PR #12332** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12332" target="_blank">Graphite</a> 
    * **PR #12018** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12018" target="_blank">Graphite</a> 
      * **PR #13006** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13006" target="_blank">Graphite</a>  👈
      * **PR #11993** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11993" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/13006?utm_source=stack-comment).

**Review by dpeng817 — CHANGES_REQUESTED** [review]
Willing to be overridden here bc I realize this is a lot of work, but I think the pattern used to update here doesn't match expected convention - I think the modern equivalent of a daily schedule is a job with a PartitionsDefinition + creating a schedule from that job - @sryza just to make sure we're in agreement there, but if so I think it's worth fixing - when people see this pattern, they'll probably repeat it in tests, eventually someone will recommend it to a user by accident, etc.

**Comment by smackesey** [comment]
> think the pattern used to update here doesn't match expected convention - I think the modern equivalent of a daily schedule is a job with a PartitionsDefinition + creating a schedule from that job

Hmm definitely to updating the pattern where relevant (as I think it is in the internal comment), but I don't think it is in many cases? In many of the places `@daily_schedule` is used in tests (e.g. in `test_scheduler_run.py`), it's just to get a schedule that runs every day-- the partitions are irrelevant to what's being tested.

>when people see this pattern, they'll probably repeat it in tests, eventually someone will recommend it to a user by accident, etc.

What is the harmful pattern? Using `@schedule` with `cron_schedule="@daily"`?

**Comment by sryza** [comment]
> I think the modern equivalent of a daily schedule is a job with a PartitionsDefinition + creating a schedule from that job

I'm not fully caught up on this PR, so take this with a grain of salt, but one thing I want to add: with our previous APIs, I think partitioned schedules were overused. I think non-partitioned daily schedules are a totally legitimate usage pattern.

File: python_modules/dagster/dagster/_core/definitions/partition.py [hunk 1]
```diff
@@ -889,212 +878,6 @@ def get_partition_names(
     ) -> Sequence[str]:
         return [part.name for part in self.get_partitions(current_time, dynamic_partitions_store)]
 
-    def create_schedule_definition(
```
Comment (dpeng817) [inline python_modules/dagster/dagster/_core/definitions/partition.py:892]: so satisfying

File: python_modules/dagster/dagster_tests/cli_tests/command_tests/test_cli_commands.py [hunk 2]
```diff
@@ -157,6 +146,11 @@ def error_config(_):
...
 
     return {
+        "partitioned_pipeline_partitions": PartitionSetDefinition(
+            name="partitioned_pipeline_partitions",
+            pipeline_name="partitioned_pipeline",
+            partition_fn=lambda: string.digits,
+        ),
         "baz_partitions": PartitionSetDefinition(
             name="baz_partitions",
             pipeline_name="baz",
```
Comment (dpeng817) [inline python_modules/dagster/dagster_tests/cli_tests/command_tests/test_cli_commands.py:156]: hrm what's the deal here - ideally there should be no `PartitionSetDefinitions` after this

File: python_modules/dagster/dagster_tests/scheduler_tests/test_scheduler_run.py [hunk 3]
```diff
@@ -882,11 +854,16 @@ def test_schedule_without_timezone(instance, executor):
 
 
 @pytest.mark.parametrize("executor", get_schedule_executors())
-def test_bad_env_fn_no_retries(instance, workspace_context, external_repo, executor):
-    external_schedule = external_repo.get_external_schedule("bad_env_fn_schedule")
+def test_bad_eval_fn_no_retries(
+    instance: DagsterInstance,
+    workspace_context: WorkspaceProcessContext,
+    external_repo: ExternalRepository,
+    executor: ThreadPoolExecutor,
```
Comment (dpeng817) [inline python_modules/dagster/dagster_tests/scheduler_tests/test_scheduler_run.py:861]: thank u for ur service w these annotations

**Review by dpeng817 — APPROVED** [review]
This looks good outside of the PartitionSetDefinition stuff - is that still necessary?

File: python_modules/dagster/dagster_tests/cli_tests/command_tests/test_cli_commands.py (hunk 2 above)
Comment (smackesey) [reply python_modules/dagster/dagster_tests/cli_tests/command_tests/test_cli_commands.py:156]: This one is grouped with all the others that were already there and I'm taking aim at all PartitionsSetDefinition in the next PR in this stack

**Pull request merged**